javascript/modules/client/src/internal/constants/encoding.ts

```diff
@@ -0,0 +1,5 @@
+export const AVAILABLE_FUNCTION_SIGNATURES: string[] = [
+  "function changeVoteConfig(uint64,uint64,uint64)",
+  "function withdraw(address,address,uint256,string)",
+  "function setMetadata(bytes)",
+];
```

Comment:
Yes, we can do something like this for each function:

```ts
const daoInterface = DAO__factory.createInterface();
const withdrawFunction = daoInterface.getFunction("withdraw");
withdrawFunction.format('minimal')
```

Does not make much sense because we are hardcoding the name of the function anyway, so if the function name changes we will need to change it

Reply:
Right, but if one of the parameters changes, we will get the updated hash, for free. And if the function no longer exists, the call will throw an Error and we will immediately realize it. Also, for free